### ISSUE TITLE
Remove and replace jQuery and Bootstrap JS

### ISSUE THREAD

**Issue by askvortsov1** [flarum/core#2345]
Followup to https://github.com/flarum/core/issues/1821

Right now, we're considering:

* [ ]  jquery => cash (Typescript, better maintained, smaller) + velocity (separate animation library)
* [ ]  bootstrap => micromodal, tooltipjs, custom solutions

**Comment by askvortsov1** [comment]
Relevant: https://github.com/flarum/core/issues/2303#issuecomment-751206229 (replacing bootstrap modals with something better, or a vanilla solution).

**Comment by askvortsov1** [comment]
> Relevant: [#2303 (comment)](https://github.com/flarum/core/issues/2303#issuecomment-751206229) (replacing bootstrap modals with something better, or a vanilla solution would fix that issue).

**Comment by askvortsov1** [comment]
Should also strive to use vanilla JS wherever possible to make the eventual transition simpler.

**Comment by davwheat** [comment]
Have you considered using [`clsx`](https://www.npmjs.com/package/clsx) as opposed to `classnames`? It's a drop-in replacement and far, far smaller.

**Comment by askvortsov1** [comment]
If it works, I'm all for it!

**Comment by askvortsov1** [comment]
@datitisev I think we should also include your idea of tooltips as a composable component as part of this. Do we have a separate issue for that?

**Comment by datitisev** [comment]
@askvortsov1 I don't think we have a separate issue; I don't remember making one & wasn't able to find one.

**Comment by davwheat** [comment]
I'm going to try and push for modals to use the native `<dialog>` element as opposed to any custom JS-based solutions.

[It's supported in Firefox and Chrome for Android](https://caniuse.com/dialog), and [Google even made their own polyfill](https://github.com/GoogleChrome/dialog-polyfill) which works with their desktop version of Chrome.

Check out the demos to see what's possible: https://googlechrome.github.io/dialog-polyfill/

**Comment by SychO9** [comment]
I love that CSS is powerful enough to do things like tooltips, but the problem with this solution is that it uses psuedo elements to achieve that, quoting:

> Drawbacks
Balloon.css make use of pseudo-elements thus self-closing elements such as `<img>`, `<input>` and `<hr>` will not render tooltips.
>
> Also keep in mind that if pseudo elements are already in use on an element, the tooltip will conflict with them resulting in potential bugs.

**Comment by davwheat** [comment]
Any issue arising from the use of pseudoelements could be resolved immediately by wrapping the node in a plain div or span.

**Comment by askvortsov1** [comment]
> @datitisev I think we should also include your idea of tooltips as a composable component as part of this. Do we have a separate issue for that?
> 

A tool tip helper component has been suggested and I think it's a great idea to promote consistency in UI and abstract away the implementation.
